packages/truffle-core/lib/test.js

```diff
@@ -144,36 +145,33 @@ const Test = {
     return web3.eth.getAccounts();
   },
 
-  compileContractsWithTestFilesIfNeeded: function(
+  compileContractsWithTestFilesIfNeeded: async function(
     solidityTestFiles,
     config,
     testResolver
   ) {
-    return new Promise(function(accept, reject) {
-      Profiler.updated(config.with({ resolver: testResolver }), function(
-        err,
-        updated
-      ) {
-        if (err) return reject(err);
-
-        updated = updated || [];
-
-        const compileConfig = config.with({
-          all: config.compileAll === true,
-          files: updated.concat(solidityTestFiles),
-          resolver: testResolver,
-          quiet: config.quiet,
-          quietWrite: true
-        });
-        // Compile project contracts and test contracts
-        Contracts.compile(compileConfig)
-          .then(result => {
-            const paths = result.outputs.solc;
-            accept(paths);
-          })
-          .catch(reject);
-      });
+    const updated =
+      (await promisify(Profiler.updated)(
+        config.with({ resolver: testResolver })
+      )) || [];
+
+    const compileConfig = config.with({
+      all: config.compileAll === true,
+      files: updated.concat(solidityTestFiles),
+      resolver: testResolver,
+      quiet: false,
```

Comment:
Don't we want this to come from the config? Wasn't this something changed recently by @CruzMolina?

Reply:
Oh yeah you're right!